### PR TITLE
feat: `sponsorblock` patch

### PR DESCRIPTION
**Pull request by oSumAtrIX**
Will depend on:
- https://github.com/revanced/revanced-integrations/issues/79

Todo:
- https://github.com/revanced/revanced-patches/pull/175#issuecomment-1186047965

### CODE REVIEW

**Comment by Sculas** [comment]
Any updates on this patch?

**Comment by oSumAtrIX** [comment]
It is being worked on.

File: src/main/resources/sponsorblock/values/strings.xml [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+    <string name="segments_sponsor">Sponsor</string>
+    <string name="segments_sponsor_sum">Paid promotion, paid referrals and direct advertisements</string>
+    <string name="segments_intermission">Intermission/Intro Animation</string>
+    <string name="segments_intermission_sum">An interval without actual content. Could be a pause, static frame, repeating animation</string>
+    <string name="segments_endcards">Endcards/Credits</string>
+    <string name="segments_endcards_sum">Credits or when the YouTube endcards appear. Not spoken conclusions</string>
+    <string name="segments_subscribe">Interaction Reminder (Subscribe)</string>
+    <string name="segments_subscribe_sum">When there is a short reminder to like, subscribe or follow them in the middle of content</string>
+    <string name="segments_selfpromo">Unpaid/Self Promotion</string>
+    <string name="segments_selfpromo_sum">Similar to "sponsor" except for unpaid or self promotion. This includes sections about merchandise, donations, or information about who they collaborated with</string>
```
Comment (mchangrh) [inline src/main/resources/sponsorblock/values/strings.xml:38]:
```suggestion
    <string name="segments_selfpromo_sum">When there is unpaid or self promotion. This includes specific sections about merchandise, donations, or information about who they collaborated with</string>
```

File: src/main/resources/sponsorblock/values/strings.xml [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+
+    <string name="action_segments">Segments</string>
+    <string name="action_browser">SB Browser</string>
+
+    <string name="api_url_changed">API URL changed</string>
+    <string name="api_url_reset">API URL reset</string>
+    <string name="api_url_invalid">Provided API URL is invalid</string>
+
+    <string name="revanced_sb">SponsorBlock</string>
+    <string name="revanced_sb_firstrun">Are you aware of the SponsorBlock integration in Vanced? With it you can skip sponsored segments in the videos. You can enable/disable this in the settings at any time.</string>
```
Comment (mchangrh) [inline src/main/resources/sponsorblock/values/strings.xml:183]:
```suggestion
    <string name="revanced_sb_firstrun">Are you aware of the SponsorBlock integration in ReVanced? With it you can skip sponsored segments in the videos. You can enable/disable this in the settings at any time.</string>
```

File: src/main/resources/sponsorblock/values/strings.xml [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+    <string name="revanced_old_style_quality_settings_summary_on">Using old style video quality settings</string>
+
+    <string name="general_time_without_sb">Show time without segments</string>
+    <string name="general_time_without_sb_sum">This time appears in brackets next to the current time. This shows the total video duration minus any segments.</string>
+    <string name="general_whitelisting">Channel whitelisting</string>
+    <string name="general_whitelisting_sum">Use the Segments button under the player to whitelist a channel</string>
+    <string name="general_browser_button">Enable SB Browser button</string>
+    <string name="general_browser_button_sum">Clicking this button under the player will open sb.ltn.fi where you can see more details about segments.</string>
+    <string name="segments_preview">Preview/Recap</string>
+    <string name="segments_preview_sum">Quick recap of previous episodes, or a preview of what\'s coming up later in the current video. Meant for edited together clips, not for spoken summaries.</string>
```
Comment (mchangrh) [inline src/main/resources/sponsorblock/values/strings.xml:152]:
```suggestion
    <string name="segments_preview_sum">Recap of previous episodes, or a preview of what\'s coming up later in the current video or future videos in the same series. Intended for edited together clips that do not provide additional information.</string>
```

File: src/main/resources/sponsorblock/values/strings.xml [hunk 4]
```diff
@@ -0,0 +1,184 @@
...
+    <string name="new_segment_confirm_title">Are the times correct?</string>
+    <string name="new_segment_confirm_content" formatted="false">The segment lasts from %02d:%02d to %02d:%02d (%d minutes %02d seconds)\nIs it ready to submit?</string>
+    <string name="new_segment_mark_locations_first">Mark two locations on the time bar first</string>
+    <string name="new_segment_edit_by_hand_title">Edit timing of segment manually</string>
+    <string name="new_segment_edit_by_hand_content">Do you want to edit the timing for the start or end of the segment?</string>
+    <string name="new_segment_edit_by_hand_saved">Done</string>
+    <string name="new_segment_edit_by_hand_parse_error">Invalid time given</string>
+
+    <string name="sb_guidelines_preference_title">View guidelines</string>
+    <string name="sb_guidelines_preference_sum">Guidelines contain tips about submitting a segment</string>
```
Comment (mchangrh) [inline src/main/resources/sponsorblock/values/strings.xml:101]:
```suggestion
    <string name="sb_guidelines_preference_sum">Guidelines contain tips and rules about submitting segments</string>
```

File: src/main/resources/sponsorblock/values/strings.xml [hunk 5]
```diff
@@ -0,0 +1,184 @@
...
+    <string name="segments_intermission">Intermission/Intro Animation</string>
+    <string name="segments_intermission_sum">An interval without actual content. Could be a pause, static frame, repeating animation</string>
+    <string name="segments_endcards">Endcards/Credits</string>
+    <string name="segments_endcards_sum">Credits or when the YouTube endcards appear. Not spoken conclusions</string>
+    <string name="segments_subscribe">Interaction Reminder (Subscribe)</string>
+    <string name="segments_subscribe_sum">When there is a short reminder to like, subscribe or follow them in the middle of content</string>
+    <string name="segments_selfpromo">Unpaid/Self Promotion</string>
+    <string name="segments_selfpromo_sum">Similar to "sponsor" except for unpaid or self promotion. This includes sections about merchandise, donations, or information about who they collaborated with</string>
+    <string name="segments_nomusic">Music: Non-Music Section</string>
+    <string name="segments_nomusic_sum">Only for use in music videos. This includes introductions or outros in music videos</string>
```
Comment (mchangrh) [inline src/main/resources/sponsorblock/values/strings.xml:40]:
```suggestion
    <string name="segments_nomusic_sum">Only for use in music videos. This includes any content not in the Spotify Release</string>
```

File: src/main/resources/sponsorblock/values/strings.xml [hunk 6]
```diff
@@ -0,0 +1,184 @@
...
+    <string name="general_cache_sum">Frequently watched videos (eg. music videos) may store segments in app cache to make skipping segments faster</string>
+    <string name="general_cache_clear">Clear SponsorBlock segments cache</string>
+    <string name="segments_sponsor">Sponsor</string>
+    <string name="segments_sponsor_sum">Paid promotion, paid referrals and direct advertisements</string>
+    <string name="segments_intermission">Intermission/Intro Animation</string>
+    <string name="segments_intermission_sum">An interval without actual content. Could be a pause, static frame, repeating animation</string>
+    <string name="segments_endcards">Endcards/Credits</string>
+    <string name="segments_endcards_sum">Credits or when the YouTube endcards appear. Not spoken conclusions</string>
+    <string name="segments_subscribe">Interaction Reminder (Subscribe)</string>
+    <string name="segments_subscribe_sum">When there is a short reminder to like, subscribe or follow them in the middle of content</string>
```
Comment (mchangrh) [inline src/main/resources/sponsorblock/values/strings.xml:36]:
```suggestion
    <string name="segments_subscribe_sum">When there is a short reminder to like, subscribe, follow or interact with them on any paid or free platform</string>
```

File: src/main/resources/sponsorblock/values/strings.xml [hunk 7]
```diff
@@ -0,0 +1,184 @@
...
+    <string name="new_segment_mark_locations_first">Mark two locations on the time bar first</string>
+    <string name="new_segment_edit_by_hand_title">Edit timing of segment manually</string>
+    <string name="new_segment_edit_by_hand_content">Do you want to edit the timing for the start or end of the segment?</string>
+    <string name="new_segment_edit_by_hand_saved">Done</string>
+    <string name="new_segment_edit_by_hand_parse_error">Invalid time given</string>
+
+    <string name="sb_guidelines_preference_title">View guidelines</string>
+    <string name="sb_guidelines_preference_sum">Guidelines contain tips about submitting a segment</string>
+    <string name="sb_guidelines_popup_title">There are guidelines</string>
+    <string name="sb_guidelines_popup_content">It\'s recommended to read the Sponsor Block guidelines before submitting any segment</string>
```
Comment (mchangrh) [inline src/main/resources/sponsorblock/values/strings.xml:103]:
```suggestion
    <string name="sb_guidelines_popup_content">It\'s recommended to read the SponsorBlock guidelines before submitting any segment</string>
```

File: src/main/resources/sponsorblock/values/strings.xml [hunk 8]
```diff
@@ -0,0 +1,184 @@
...
+    <string name="revanced_old_style_quality_settings_title">Quality Settings style</string>
+    <string name="revanced_old_style_quality_settings_summary_off">Using default style video quality settings</string>
+    <string name="revanced_old_style_quality_settings_summary_on">Using old style video quality settings</string>
+
+    <string name="general_time_without_sb">Show time without segments</string>
+    <string name="general_time_without_sb_sum">This time appears in brackets next to the current time. This shows the total video duration minus any segments.</string>
+    <string name="general_whitelisting">Channel whitelisting</string>
+    <string name="general_whitelisting_sum">Use the Segments button under the player to whitelist a channel</string>
+    <string name="general_browser_button">Enable SB Browser button</string>
+    <string name="general_browser_button_sum">Clicking this button under the player will open sb.ltn.fi where you can see more details about segments.</string>
```
Comment (mchangrh) [inline src/main/resources/sponsorblock/values/strings.xml:150]:
```suggestion
    <string name="general_browser_button_sum">Clicking this button under the player will open sb.ltn.fi where you can see all the segments on the video.</string>
```

File: src/main/resources/sponsorblock/values/strings.xml [hunk 9]
```diff
@@ -0,0 +1,184 @@
...
+    <string name="segments_endcards">Endcards/Credits</string>
+    <string name="segments_endcards_sum">Credits or when the YouTube endcards appear. Not spoken conclusions</string>
+    <string name="segments_subscribe">Interaction Reminder (Subscribe)</string>
+    <string name="segments_subscribe_sum">When there is a short reminder to like, subscribe or follow them in the middle of content</string>
+    <string name="segments_selfpromo">Unpaid/Self Promotion</string>
+    <string name="segments_selfpromo_sum">Similar to "sponsor" except for unpaid or self promotion. This includes sections about merchandise, donations, or information about who they collaborated with</string>
+    <string name="segments_nomusic">Music: Non-Music Section</string>
+    <string name="segments_nomusic_sum">Only for use in music videos. This includes introductions or outros in music videos</string>
+    <string name="segments_filler">Filler Tangent/Jokes</string>
+    <string name="segments_filler_sum">Tangential scenes added only for filler or humor that are not required to understand the main content of the video. This should not include segments providing context or background details</string>
```
Comment (mchangrh) [inline src/main/resources/sponsorblock/values/strings.xml:42]:
```suggestion
    <string name="segments_filler_sum">Tangential scenes added only for filler or humor that are not required to understand the main content of the video. This should not include context or background details</string>
```

File: src/main/resources/sponsorblock/values/strings.xml [hunk 10]
```diff
@@ -0,0 +1,184 @@
...
+    <string name="settings_export_failed">Failed to export settings</string>
+    <string name="general_cache">Cache segments locally</string>
+    <string name="general_cache_sum">Frequently watched videos (eg. music videos) may store segments in app cache to make skipping segments faster</string>
+    <string name="general_cache_clear">Clear SponsorBlock segments cache</string>
+    <string name="segments_sponsor">Sponsor</string>
+    <string name="segments_sponsor_sum">Paid promotion, paid referrals and direct advertisements</string>
+    <string name="segments_intermission">Intermission/Intro Animation</string>
+    <string name="segments_intermission_sum">An interval without actual content. Could be a pause, static frame, repeating animation</string>
+    <string name="segments_endcards">Endcards/Credits</string>
+    <string name="segments_endcards_sum">Credits or when the YouTube endcards appear. Not spoken conclusions</string>
```
Comment (mchangrh) [inline src/main/resources/sponsorblock/values/strings.xml:34]:
```suggestion
    <string name="segments_endcards_sum">Credits or when the YouTube endcards appear. Not for spoken conclusions</string>
```

File: src/main/resources/sponsorblock/values/strings.xml [hunk 11]
```diff
@@ -0,0 +1,184 @@
...
+    <string name="diff_segments">What to do with different segments</string>
+    <string name="general">General</string>
+    <string name="general_skiptoast">Show a toast when skipping segment automatically</string>
+    <string name="general_skiptoast_sum">Click to see an example toast</string>
+    <string name="general_skipcount">Skip count tracking</string>
+    <string name="general_skipcount_sum">This lets SponsorBlock leaderboard system know how much time people have saved. The extension sends a message to the server each time you skip a segment</string>
+    <string name="general_adjusting">Adjusting new segment step</string>
+    <string name="general_adjusting_sum">This is the number of milliseconds you can move when you use the time adjustment buttons while adding new segment</string>
+    <string name="general_min_duration">Minimum segment duration</string>
+    <string name="general_min_duration_sum">Segments shorter than the set value (in seconds) will not be skipped or show in the player</string>
```
Comment (mchangrh) [inline src/main/resources/sponsorblock/values/strings.xml:16]:
```suggestion
    <string name="general_min_duration_sum">Segments shorter than the set value (in seconds) will not be skipped or shown in the player</string>
```

**Review by mchangrh — CHANGES_REQUESTED** [review]
Additional requested changes
- handler for 403 responses
  This usually contains a warning for the user or a message explaining why the category is not accepting submissions.

- Force refresh before submitting if cache is enabled - This might mislead the user into thinking that a video doesn't have a segment and submitting a bad duplicate

misc:
if you need videos for testing, https://wiki.sponsor.ajay.app/w/Testing#Demo_Videos has a good selection (in interesting cases too)

**Comment by mchangrh** [comment]
Sorry about that, I thought that GH allowed accepting all changes in one commit. I'll make a PR in future.

**Comment by oSumAtrIX** [comment]
> Sorry about that, I thought that GH allowed accepting all changes in one commit. I'll make a PR in future.

**Pull request merged**